src/screens/FeedScreen/hooks/useFeedUtilsHook.ts

```diff
@@ -0,0 +1,31 @@
+import {NativeScrollEvent, NativeSyntheticEvent} from 'react-native';
+
+import dimen from '../../../utils/dimen';
+
+const useFeedUtilsHook = () => {
+  const getCurrentPostViewed = (momentumEvent: NativeSyntheticEvent<NativeScrollEvent>): number => {
+    const {y} = momentumEvent.nativeEvent.contentOffset;
+    const shownIndex = Math.ceil(y / dimen.size.FEED_CURRENT_ITEM_HEIGHT);
+    return shownIndex;
+  };
+
+  const isSamePostViewed = (
+    momentumEvent: NativeSyntheticEvent<NativeScrollEvent>,
+    viewPostTimeIndex: number
+  ): boolean => {
+    console.log(
+      'isSamePostViewed',
+      getCurrentPostViewed(momentumEvent),
+      viewPostTimeIndex,
+      getCurrentPostViewed(momentumEvent) === viewPostTimeIndex
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `console.log` statement in the `isSamePostViewed` function could potentially leak sensitive information about the user's interactions. It's generally a good practice to avoid logging sensitive data or user interactions. If this log is necessary for debugging, consider using a feature flag to control its execution in non-production environments.
```diff
    console.log(
      'isSamePostViewed',
      getCurrentPostViewed(momentumEvent),
      viewPostTimeIndex,
      getCurrentPostViewed(momentumEvent) === viewPostTimeIndex
    );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->